switchtec.c

```diff
@@ -1004,6 +1020,9 @@ static int ioctl_event_ctl(struct switchtec_dev *stdev,
 	if (ctl.flags & SWITCHTEC_IOCTL_EVENT_FLAG_UNUSED)
 		return -EINVAL;
 
+	if (!event_supported(stdev, ctl.event_id, ctl.index))
```

Comment:
This definitely seems like the wrong place for this. It ignores EVENT_IDX_ALL and it's an extra read that's not needed. Just move the check into event_ctl() where it already reads the hdr.

You'll probably need to also skip the specific error when doing all events.

Reply:
> You'll probably need to also skip the specific error when doing all events.

Sorry, I didn't get your point.

Reply:
My point was if the user is doing an IDX_ALL ioctl, you don't want a single not supported event to stop the loop.